pkg/scheduler/algorithm/priorities/resource_bin_packing.go

```diff
@@ -0,0 +1,240 @@
+				if requestedInit.Memory < memory {
+					requestedInit.Memory = memory
+				}
+			}
+			if requestedInit.Memory > requested.Memory {
+				requested.Memory = requestedInit.Memory
+			}
+			//println("Memory used: %d, requested: %d, available: %d",
+			//	nodeInfo.NonZeroRequest().Memory, requested.Memory, allocatable.Memory)
+			return float64((nodeInfo.NonZeroRequest().Memory+requested.Memory)*schedulerapi.MaxPriority) / float64(allocatable.Memory)
+		}
+
+	case v1.ResourceEphemeralStorage:
+		{
+			for i := range pod.Spec.Containers {
+				container := &pod.Spec.Containers[i]
+				if quantity, ok := container.Resources.Requests[v1.ResourceEphemeralStorage]; ok {
+					requested.EphemeralStorage += quantity.Value()
+				}
+
+			}
+			for i := range pod.Spec.InitContainers {
+				container := &pod.Spec.InitContainers[i]
+				if quantity, ok := container.Resources.Requests[v1.ResourceEphemeralStorage]; ok {
+					if requestedInit.EphemeralStorage < quantity.Value() {
+						requestedInit.EphemeralStorage = quantity.Value()
+					}
+				}
+			}
+			if requestedInit.EphemeralStorage > requested.EphemeralStorage {
+				requested.EphemeralStorage = requestedInit.EphemeralStorage
+			}
+			return float64((used.EphemeralStorage+requested.EphemeralStorage)*schedulerapi.MaxPriority) / float64(allocatable.EphemeralStorage)
+
+		}
+	default:
+		if v1helper.IsScalarResourceName(resource) {
+			if podRequestsResource(*pod, resource) {
+				for i := range pod.Spec.Containers {
+					container := &pod.Spec.Containers[i]
+					if quantity, ok := container.Resources.Requests[resource]; ok {
+						requested.AddScalar(resource, quantity.Value())
+					}
+				}
+				for i := range pod.Spec.InitContainers {
+					container := &pod.Spec.InitContainers[i]
+					if quantity, ok := container.Resources.Requests[resource]; ok {
+						if requestedInit.ScalarResources[resource] < quantity.Value() {
+							requested.SetScalar(resource, quantity.Value())
+						}
+					}
+
+				}
+				if requestedInit.ScalarResources[resource] > requested.ScalarResources[resource] {
+					requested.ScalarResources[resource] = requestedInit.ScalarResources[resource]
+				}
+
+				return float64((used.ScalarResources[resource]+requested.ScalarResources[resource])*schedulerapi.MaxPriority) / float64(allocatable.ScalarResources[resource])
+
+			}
+		}
+
+	}
+	return 0
+}
+
+// podRequestsResource checks if a given pod requests the scare resource. if false the priority is set to 0
+func podRequestsResource(pod v1.Pod, resource v1.ResourceName) bool {
+	if strings.Contains(string(resource), "cpu") {
+		return true
+	}
+	if strings.Contains(string(resource), "memory") {
+		return true
+	}
+	containerRequestsResource := func(container v1.Container) bool {
+		for resName, quantity := range container.Resources.Requests {
+			if resName == resource && quantity.MilliValue() > 0 {
+				return true
+			}
+		}
+		for resName, quantity := range container.Resources.Limits {
```

Comment:
do we need to consider limits?

Reply:
yea, I think only requests was needed. Updated

Reply:
yes the limits are never considered during scheduling